tests/v2/validation/v2prov/encryption_key_rotation_test.go

```diff
@@ -170,16 +184,46 @@ func (r *V2ProvEncryptionKeyRotationTestSuite) TestEncryptionKeyRotation() {
 	setEncryptSecret(r.T(), r.client, id)
 
 	prefix := "encryption-key-rotation-"
+
+	r.Run(prefix+"new-cluster", func() {
+		rotateEncryptionKeys(r.T(), r.client, id, 10*time.Minute)
+	})
+
+	if r.client.Flags.GetValue(environmentflag.Long) {
+		// create 10k secrets for stress test, takes ~30 minutes
+		createSecretsForCluster(r.T(), r.client, id, totalSecrets)
+
+		r.Run(prefix+"stress-test", func() {
+			rotateEncryptionKeys(r.T(), r.client, id, 1*time.Hour) // takes ~45 minutes for HA
+		})
+	}
+}
+
+func (r *V2ProvEncryptionKeyRotationTestSuite) TestEncryptionKeyRotationWithVaiEnabled() {
+	subSession := r.session.NewSession()
+	defer subSession.Cleanup()
+
+	id, err := clusters.GetV1ProvisioningClusterByName(r.client, r.clusterName)
+	require.NoError(r.T(), err)
+
+	setEncryptSecret(r.T(), r.client, id)
+	enableVai := true
+	var once sync.Once
```

Comment:
again, I'm confused on what this does. sync.Once can just be nil? if so, why not just instantiate it in the EnableVAI function itself instead of passing it along as a var?